frontend/src/App.tsx

```diff
@@ -1,35 +1,61 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Route, BrowserRouter as Router, Routes } from 'react-router-dom';
 import { ToastContainer } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 
+import 'slick-carousel/slick/slick-theme.css';
+import 'slick-carousel/slick/slick.css';
+
 import './App.css';
 import { Header } from './header/Header.component';
 import { useTheme } from './hooks/useTheme';
+import { AuthModal } from './modalWindow/AuthModal';
 import { ProductsComponent } from './product/Products.component';
 import { ProductPage } from './productPages/productPage.component';
-import { SignIn } from './signIn/SingIn.component';
 import { SignOut } from './signOut/SignOut.component';
-import { Signup } from './signUp/SignUp.component';
-import { TokenProvider } from './tokenContext/TokenContext';
+import { TokenProvider, useTokenContext } from './tokenContext/TokenContext';
 
-function App() {
+function AppContent() {
   const { theme, toggleTheme } = useTheme();
+  const { token } = useTokenContext();
+  const [isAuthModalOpen, setIsAuthModalOpen] = React.useState(false);
+  const [isSignIn, setIsSignIn] = React.useState(true);
```

Comment:
Remove React import, and get only required imports from it 

```suggestion
  const [isAuthModalOpen, setIsAuthModalOpen] = useState(false);
  const [isSignIn, setIsSignIn] = useState(true);
```